src/e3/anod/qualifiers_manager.py

```diff
@@ -695,7 +696,7 @@ def __generate_qualifier_part(
                 return "_" + qualifier_name + "-" + qualifier_value
         else:
             raise AnodError(
```

Comment:
All those are qualifier errors I guess (Thanks Leo for pinpointing it). If so, raise a `QualifierError` and update the tests too :)
To me, all error message mentionning that an unexpected qualifier was encountered should raise a `QualifierError`.

Reply:
I forgot to add a comment to describe what is QualifierError.

Let's discuss it here :)

For me there are two kind of error raised by the QualifiersManager:
-  AnodError: The error is static (structural) and come from a mistake in the spec itself
-  QualifierError: The error is dynamic, it lies in the qualifiers passed to the spec, not in the spec itself. The culprit is the caller.

That is why I am only adding the context information to the QualifierErrors

What do you  think ?

Reply:
The one you are pointing at is fishy. To produce it, you have to mess up with the inner data structures of the QualifiersManager so for me it is an AnodError

Reply:
That's ok, I have read the various uses of the `__error` call or the `AnodError`. I hav no objection on you point of view :)
Now, a little comment in the `AnodError` case to say why we do not raise `QualifierError` could be good, but still, it's only my point of view.

Reply:
I have added a Little comment in the QualifiersManager class. That is what you meant ?

Reply:
Ok